server/controllers/auth/login.ts

```diff
@@ -0,0 +1,52 @@
+import { NextFunction, Request, Response } from 'express';
+import { compare } from 'bcryptjs';
+import { loginSchema, CustomedError, signToken } from '../../utils';
+import { Donor } from '../../database/models';
+
+const login = async (req: Request, res: Response, next: NextFunction) => {
+  const { email, password } = req.body;
+  try {
+    await loginSchema.validateAsync({ email, password });
+
+    const user = await Donor.findOne({
+      where: {
+        email,
+      },
+    });
+    if (!user) {
+      throw new CustomedError(
+        "Email doesn't exists, Try another one or sign up",
+        400,
+      );
+    }
+
+    const isPasswordValidate = await compare(
+      password,
+      user?.getDataValue('password'),
+    );
+    if (!isPasswordValidate) {
+      throw new CustomedError('Incorrect password, please try again', 400);
+    }
+
+    const payload = {
+      id: user?.getDataValue('id'),
+      name: user?.getDataValue('name'),
+      isAdmin: user?.getDataValue('is_admin'),
+    };
+    const token = await signToken(payload);
+    if (typeof token !== 'string') {
+      throw new CustomedError('Unexpected Error', 500);
+    }
```

Comment:
Why did you need to add this condition?

Reply:
Because if there is error when sign token, this condition will throw it

Reply:
But if there is an error then it will be catched in the catch.

Reply:
Yes, that's more efficient than handle error by another property.

Reply:
Repaired.

Reply:
Yes, it will caught automatically.